internal/template-validator/virtinformers/vmcache.go

```diff
@@ -231,7 +231,7 @@ func (v *vmCache) Replace(list []interface{}, _ string) error {
 	}
 
 	v.lock.Lock()
```

Comment:
Don't know the exact use of this mutex but it seems suspicious? It didn't do anything before? (immediate unlock after lock)

Reply:
@akrejcir do you recall why this was included originally?

FWIW the `staticcheck`  plugin fails without the `defer` with the following justification https://staticcheck.io/docs/checks#SA2001

```
$ make lint
[...]
internal/template-validator/virtinformers/vmcache.go:234:2: SA2001: empty critical section (staticcheck)
        v.lock.Unlock()
        ^
make: *** [Makefile:323: lint] Error 1
```

Reply:
@akrejcir just a reminder, if you have time ^

Reply:
That is a bug, I forgot the `defer`, thanks. The critical section are the lines below it.

Reply:
https://github.com/kubevirt/ssp-operator/pull/514

Reply:
Not sure why that needs another PR when it's literally fixed here as well?

Reply:
Sorry, I didn't read this PR, only this comment.

Reply:
No issues.